inference-engine/src/multi_device/multi_device_exec_network.cpp

```diff
@@ -60,71 +90,243 @@ MultiDeviceExecutableNetwork::MultiDeviceExecutableNetwork(const DeviceMap<Infer
-            workerRequest._inferRequest->SetCallback(
-                [workerRequestPtr, this, device, idleWorkerRequestsPtr] (std::exception_ptr exceptionPtr) mutable {
-                    IdleGuard idleGuard{workerRequestPtr, *idleWorkerRequestsPtr};
-                    workerRequestPtr->_exceptionPtr = exceptionPtr;
-                    {
-                        auto capturedTask = std::move(workerRequestPtr->_task);
-                        capturedTask();
-                    }
-                    // try to return the request to the idle list (fails if the overall object destruction has began)
-                    if (idleGuard.Release()->try_push(workerRequestPtr)) {
-                        // let's try to pop a task, as we know there is at least one idle request, schedule if succeeded
-                        // if no device-agnostic tasks, let's try pop the device specific task, schedule if succeeded
-                        Task t;
-                        if (_inferPipelineTasks.try_pop(t))
-                            ScheduleToWorkerInferRequest(std::move(t));
-                        else if (_inferPipelineTasksDeviceSpecific[device]->try_pop(t))
-                            ScheduleToWorkerInferRequest(std::move(t), device);
-                    }
-                });
+            _networkFirstReady = _cpuFuture.get();
+        } catch (const std::exception& e) {
+            printf("Warning: load network to CPU failed: %s\n", e.what());
+            _networkActualNeeded = _acceleratorFuture.get();
         }
+    } else if (_acceleratorFuture.valid()) {  // only accelerator is valid, like AUTO:GPU
+        _networkActualNeeded = _acceleratorFuture.get();
+    } else if (_cpuFuture.valid()) {  // only CPU is valid, like AUTO:CPU
+        _networkActualNeeded = _cpuFuture.get();
+    } else {
+        IE_THROW() << "No device task available";
+    }
+
+    // if only one device or cpu device faild, so the ActualNetwork is ok
+    if (!_acceleratorFuture.valid()) {
+        _alreadyActualNetwork = true;
+        // setPerfcounts
+        SetPerfCounts();
+    }
+}
+
+bool MultiDeviceExecutableNetwork::IsActualNetworkReady() const {
+    if (_alreadyActualNetwork) {
+        WaitActualNetworkReady();
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void MultiDeviceExecutableNetwork::WaitActualNetworkReady() const {
+    // Maybe differnt API will call this function, so add call once here
+    // for every MultiDeviceExecutableNetwork instance
+    std::call_once(_oc, [&] () {
+            if (_acceleratorFuture.valid()) {
+                _networkActualNeeded = _acceleratorFuture.get();
+                SetPerfCounts();
+            }
+            });
+}
+
+void MultiDeviceExecutableNetwork::SetPerfCounts() const {
+    try {
+        _needPerfCounters = _networkActualNeeded->GetMetric(PluginConfigParams::KEY_PERF_COUNT).as<std::string>() == PluginConfigParams::YES;
```

Comment:
from what I see the _needPerfCounters is queried from the **actual** network, but the variable is actually used when InferRequests are created, which supposedly happens when the **first** network is loaded. 
why we need this query (assuming the needPerfCounters can come from the plugin to the constructor, as in line 82)